### PR TITLE
Add support for dumping native stacks with cdb

### PR DESCRIPTION
**Pull request by jkoritzinsky**
SOS is still TBD as we're trying to determine the best way to pull down SOS since the version of cdb that's installed is too old to have SOS included.

~NO-MERGE because this PR also introduces an intentional test timeout to validate the behavior.~

### CODE REVIEW

File: src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs [hunk 1]
```diff
@@ -691,6 +725,21 @@ public int RunTest(string executable, string outputFile, string errorFile, strin
                                         TryPrintStackTraceFromCrashReport(dmpFile.FullName, outputWriter);
                                     }
                                 }
+                                else
+                                {
+                                    var dmpFilesInfo = crashDumpFolderInfo.GetFiles("*.dmp").OrderByDescending(f => f.CreationTime);
```
Comment (hoyosjs) [inline src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs:730]: should we match against a pattern to rule out other crashes?

**Review by hoyosjs — APPROVED** [review]

File: src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs:730]: I don't think so as we have some tests that use custom entry-point modules (in particular in the COM space) that we don't want to miss. The time filter should be good enough to ensure we don't get other random dump files IMO.

File: src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs [hunk 2]
```diff
@@ -691,6 +730,21 @@ public int RunTest(string executable, string outputFile, string errorFile, strin
...
+                                    foreach (var dmpFile in dmpFilesInfo)
+                                    {
+                                        if (dmpFile.CreationTime < startTime)
+                                        {
+                                            // No new files since test started.
+                                            outputWriter.WriteLine("Finish looking for *.dmp. No new files created.");
```
Comment (danmoseley) [inline src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs:741]:
```suggestion
                                            outputWriter.WriteLine("Finished looking for *.dmp. No new files created.");
```

**Comment by jkoritzinsky** [comment]
I can confirm that this PR enables dumping native stack traces. I'll remove the artificial test failure and merge this after another CI run.

**Pull request merged**